app/controllers/orders_controller.rb

```diff
@@ -12,6 +16,9 @@ def show
   private
 
   def find_order
-    @order = Order.find(params[:id])
+    @order = current_user.orders.find(params[:id])
+    rescue ActiveRecord::RecordNotFound
+      flash[:notice] = "Please login appropriately to view that order."
```

Comment:
Yeah, the syntax is weird but that's essentially what it does. Originally I had it as an if @order.nil? thing but it was erroring out on the first line. The rescue saves it from the error.